kiwi/system/kernel.py

```diff
@@ -15,15 +15,23 @@
 # You should have received a copy of the GNU General Public License
 # along with kiwi.  If not, see <http://www.gnu.org/licenses/>
 #
-import os
 import re
-from collections import namedtuple
+import os
+from typing import (
+    NamedTuple, Optional
+)
 
 # project
 from kiwi.command import Command
 
 from kiwi.exceptions import KiwiKernelLookupError
 
+kernel_tuple = NamedTuple('kernel', [('name', str),
+                                     ('filename', str),
+                                     ('version', str)])
+xen_hyperv = NamedTuple('xen', [('filename', str),
+                                ('name', str)])
+
```

Comment:
Regarding the use of a tuple as type there is a hidden policy in kiwi to name them XX_type and use the same indentation style as everywhere else. Therefore can be above be written as

```python
kernel_type = NamedTuple(
    'kernel_type', [
        ('name', str),
        ('filename', str),
        ('version', str)
    ]
)

xen_hypervisor_type = NamedTuple(
    'xen_hypervisor_type', [
        ('filename', str),
        ('name', str)
    ]
)
```

Reply:
Done.